### DISCUSSION TITLE
follow-ups for community-containers

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#3432] [Ideas] [answered]
Follow-ups for https://github.com/nextcloud/all-in-one/discussions/1581
### earlier
- [ ] think about instance restore: should teh backup container run twice in a row to restore the setting first and then start the correct restore. Same for normal restore?
- [x] allow to add documentation link on the additional containers (male name clickable) https://github.com/nextcloud/all-in-one/pull/3478
- [ ] allow to define nextcloud_variables
- [ ] allow to define nextcloud_secrets
    - [ ] adjust https://github.com/goodspb/pdlib/issues/56 and https://github.com/matiasdelellis/facerecognition-external-model/issues/8 if both are available
- [x] https://github.com/nextcloud/all-in-one/pull/3642
- [x] Add 
	- [x] makemkv (use https://github.com/jlesage/docker-makemkv)?
	- [x] add libretranslate and integration https://github.com/LibreTranslate/LibreTranslate/blob/main/docker/Dockerfile https://github.com/LibreTranslate/LibreTranslate/blob/main/docker-compose.yml (afterwards redo https://github.com/nextcloud/translate/pull/33) https://github.com/szaimen/aio-libretranslate/pull/1 https://github.com/v1r0x/integration_libretranslate/issues/1
	- [x] vaultwarden https://github.com/nextcloud/vm/blob/9f58a314955bf402ba36b3c78f4717331cc58e6e/apps/vaultwarden.sh#L239-L255 https://github.com/nextcloud/all-in-one/pull/3455
	- [x] local ai container? Allow to upload the models via nextcloud itself https://github.com/nextcloud/all-in-one/pull/3495
	- [x] plex https://github.com/nextcloud/all-in-one/pull/3439 
	- [x] caddy with geoblocking https://github.com/porech/caddy-maxmind-geolocation https://github.com/nextcloud/all-in-one/pull/3459 https://github.com/nextcloud/all-in-one/pull/3460
	- [x] pihole https://github.com/pi-hole/docker-pi-hole/ https://github.com/nextcloud/all-in-one/pull/3456
	- [x] ~~pivpn?~~ (use fritzbox?)
	- [x] fail2ban https://github.com/nextcloud/all-in-one/pull/2637
- [ ] instead of modifying container to put a caddy import into place, should there be an option to modify a subpath that is automatically forwarded to the correct container in a subpath available on the nextcloud domain? https://github.com/nextcloud/all-in-one/pull/3192 (maybe also use that in the containers but must also work without)
    - [ ] move onlyoffice to community apps afterwards? (Also remove all onlyoffice logic from aio then) -- needs testing with jwt secret - maxbe sheet with the secret for now...

### later
- [ ] add containers
	- [x] add backup container that allows to view the backup, restore files, etc. (maybe base it on top of https://github.com/jlesage? https://github.com/jlesage/docker-baseimage-gui/issues/113
	- [ ] smbserver https://github.com/nextcloud/vm/blob/master/not-supported/smbserver.sh (maybe base it on top of https://github.com/jlesage? https://github.com/jlesage/docker-baseimage-gui/issues/113)
- [ ] add on the fly notification? Maybe get notificcations via nc into the nc container and then use the admin notify scirpt...
    - [ ] add off-shore backup container afterwards for regular backups -> needs to be able to notify of failed or succesful backup
- [x] allow to define if a secret gets shown in the aio interface and also allow to add a description for a secret and also allow to add a description of what the token does
- [ ] allow to set not defined environmental variables from the aio interface: by providing an input field and regex and maybe even description).
- [x] ~~allow to define custom cronjobs~~ already possible via nextcloud_exec_commands

### community could add
- [x] netdata (add an example json to the request)
- [x] pgadmin (add an example json to the request)
- [x] jellyfin (addn an example json to the request)
- [x] Scrutiny (addn an example json to the request)
- [x] lldap (should only set this one up, nothing more but add good docs how to configure nextcloud, maddy ans maybe even vaultwarden? to use it); interface should be limited to apache_ip_bindin, ldap not exposed on host); interface should be available via its own domain protected by geoblocking also add link to discussinoo to readme then
- [x] docker-mailserver (add an example json to the request) also add link to discussion to readme then
- [x] mail server (add an example json to the request) also add link to discussion to readme then
- [x] more?

**Comment by FlattusBlastus** [comment]
Doubleplusgood upvote for lldap. That said, I have been wondering about the nature of NextCloud. For home use and family / friends, LDAP/SSO is probably overkill. If you were trying to integrate vaultwarden, you can send user invites from your ldap via https://github.com/ViViDboarder/vaultwarden_ldap. However, you can't sync passwords automatically without Bitwarden Directory Connect. BDC partially integrated into Vaultwarden but the waters start to get deep.

I will use the VW community container but not try to centralize my user management. It will capture logins and passwords in the browser like any password manager. Plex will still be single user with invited friends that are external. I don't need to manage their accounts. For Kavita, I will have a single login as its consumption only and launch that from External Links. That about sums it up.

I will adopt the Caddy community container if I can. I'll need to add Kavita to the Caddyfile.
I'll report back after this round of install testing.

**Reply by FlattusBlastus** [reply]
Caddyfile is in the Caddy container... ack ... can only add services until the next container update. Please move Caddyfile to /data

**Reply by szaimen** [reply]
You can add further services by adjusting this file: https://github.com/szaimen/aio-caddy/blob/main/start.sh

**Reply by szaimen** [reply]
Apart from that see https://github.com/szaimen/aio-caddy/pull/9

**Comment by FlattusBlastus** [comment]
Time to completely remove OnlyOffice from Nextcloud AIO. Nextcloud Office is the official supported solution.

(I had my tribulations and struggles with this. Ultimately, OnlyOffice didn't work as well.)

**Comment by FlattusBlastus** [comment]
There won't be a choice on allowing users to upload models. You'll have to allow it. Any attempt to keep models in the container will require constant updating. Its a admin / developer nightmare.

**Comment by FlattusBlastus** [comment]
Strongly discourage adding an email server to community containers. Email admin is a constant punishment and shouldn't be attempted by mere mortals. Hosted email at the home is the only exception to my self hosting.

**Reply by szaimen** [reply]
It is rather intended as a POC for VPS instances. Indeed mail servers should not be hosted at home

**Comment by szaimen** [comment, marked as answer]
Follow-ups 2 in https://github.com/nextcloud/all-in-one/discussions/3889

**Comment by jakobroehrl** [comment]
Maybe we can add 
 - uptime_kuma https://github.com/louislam/uptime-kuma and 
 - a dashboard/startpage like https://github.com/pawelmalak/flame or https://github.com/bastienwirtz/homer

**Reply by szaimen** [reply]
> * uptime_kuma https://github.com/louislam/uptime-kuma and

Not sure how sensible this is when it runs on the same server

> * a dashboard/startpage like https://github.com/pawelmalak/flame or https://github.com/bastienwirtz/homer

this sounds more useful. Feel free to suggest it via https://github.com/nextcloud/all-in-one/discussions/new?category=ideas
